packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -159,18 +159,18 @@ export function attributeName(attr: parse5.AST.Default.Attribute): string {
 }
 
 export function getAttribute(
-    el: IRElement,
+    el: parse5.AST.Default.Element,
     pattern: string | RegExp
 ): parse5.AST.Default.Attribute | undefined {
-    return el.attrsList.find((attr) =>
+    return el.attrs.find((attr) =>
         typeof pattern === 'string'
             ? attributeName(attr) === pattern
             : !!attributeName(attr).match(pattern)
     );
 }
 
-export function removeAttribute(el: IRElement, pattern: string | RegExp): void {
-    el.attrsList = el.attrsList.filter((attr) =>
+export function removeAttribute(el: parse5.AST.Default.Element, pattern: string | RegExp): void {
+    el.attrs = el.attrs.filter((attr) =>
```

Comment:
This is modifying the original parse5 object, which I don't like much. Since we are removing the `attrsList`, all attributes in the IRNodes are "dismissed", so there is no need to keep it clean. Do we still need the `removeAttribute` function (and calls)?

Reply:
I pushed up a new change so that the parse5 node doesn't change.  I'm tracking the attributes in a set now instead of modifying the attrs on parse5 node.